### PR TITLE
Getting started guide

### PR DESCRIPTION
**Pull request by mwestphal**

### CODE REVIEW

File: doc/dev/GETTING_STARTED.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+### Install dependencies
+
+First make sure your system is up-to-date,
+then install the required dependencies using the terminal.
+
+#### Debian/Ubuntu/Mint
+
+```
+sudo apt install build-essential git git-lfs cmake libvtk9-dev
```
Comment (ShreenidhiR) [inline doc/dev/GETTING_STARTED.md:24]: how about mentioning python3  also I feel its good to mention the recommended Ubuntu version for people who use ubuntu

Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:24]: It is out of place to suggest an OS, but I can mention on which OS this was tested

Python3 is not a required dependency at all.

**Comment by mwestphal** [comment]
@Meakk @snoyer please review. I still have to test the macOS instructions though.

File: doc/dev/BUILD.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Build guide
 
 F3D uses a CMake based build system, so building F3D just requires installing
-needed dependencies, configuring and building.
+needed dependencies, configuring and building. If you are not used to such processes
+please take a look at our getting started guide.
```
Comment (Meakk) [inline doc/dev/BUILD.md:5]: Put a link to the page

File: doc/dev/GETTING_STARTED.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
+# Getting started guide
+
+This is a getting started guide, aimed to people that are not used
+to C++ programming, configuring and compiling. If you are already used
+to such processes, you may be looking for our build guide.
```
Comment (Meakk) [inline doc/dev/GETTING_STARTED.md:5]: Link to the build guide

File: doc/dev/GETTING_STARTED.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+This guide will help you to compile your own version of F3D with required dependencies.
+Optional dependencies or covered in the build guide.
+
+ - If you are using [Linux]
```
Comment (Meakk) [inline doc/dev/GETTING_STARTED.md:10]: Is it a valid link or do you need something like `[Linux](#Linux)`?

File: doc/dev/GETTING_STARTED.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+### Install brew
+
+Open a terminal.
+
+```
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+TODO Check if XCode is needed
```
Comment (Meakk) [inline doc/dev/GETTING_STARTED.md:127]: You don't need xcode. There is a terminal command to install developer tools

File: doc/dev/GETTING_STARTED.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+Open a terminal.
+
+```
+mkdir f3d
+cd f3d
+mkdir build
+git clone https://github.com/f3d-app/f3d src
+cd build
+cmake ../src
+make
```
Comment (snoyer) [inline doc/dev/GETTING_STARTED.md:53]: For beginners you could skip the whole out-of-tree propriety to simplify a bit
```suggestion
git clone https://github.com/f3d-app/f3d
cd f3d
mkdir build
cd build
cmake ..
make
```
But I get the argument of doing it right from the start too so not a big deal either way

File: doc/dev/GETTING_STARTED.md (hunk 5 above)
Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:127]: actually needed, brew tells me:

`A full installation of XCode is needed to install to compile this software...`

when I run `brew install vtk`. Ill need to start from scratch, not enough space to install XCode

Comment (Meakk) [reply doc/dev/GETTING_STARTED.md:127]: Even if you are using Ninja generator?

Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:127]: Can you change the generator used by brew ? In any case, I doubt that installing ninja is easier than installing xcode and a beginner user.

Comment (Meakk) [reply doc/dev/GETTING_STARTED.md:127]: You need xcode to install brew itself?

Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:127]: I need either Command Line Tools or XCode to install brew.

I need XCode to install specific packages has their are flagged as such in brew system.

File: doc/dev/GETTING_STARTED.md (hunk 6 above)
Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:53]: I cant suggest in source build dir for a software you want to modify or contribute to.

File: doc/dev/GETTING_STARTED.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+```sh
+cd C:/dev
+git clone https://github.com/Microsoft/vcpkg.git
+```
+
+ * Open cmd
+
+```sh
+cd C:\dev
```
Comment (Meakk) [inline doc/dev/GETTING_STARTED.md:97]: Not a big deal, but let's be consistent with the previous command

```suggestion
cd C:/dev
```

File: doc/dev/GETTING_STARTED.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+Note: Last command will take a while. It download, compile and install all dependencies, including
+many optional ones.
+
+#### Build F3D
+
+ * Open cmd
+
+```sh
+cd C:\dev\f3d\build
```
Comment (Meakk) [inline doc/dev/GETTING_STARTED.md:110]: same here

**Review by Meakk — APPROVED** [review]
Minor changes, but otherwise it looks clear to me.

File: doc/dev/GETTING_STARTED.md (hunk 7 above)
Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:97]: cmd and git bash do not use the same syntax afaik.

File: doc/dev/GETTING_STARTED.md (hunk 8 above)
Comment (mwestphal) [reply doc/dev/GETTING_STARTED.md:110]: same

File: doc/dev/GETTING_STARTED.md (hunk 7 above)
Comment (Meakk) [reply doc/dev/GETTING_STARTED.md:97]: fair enough

**Pull request merged**